### PR TITLE
Feature/finish testing

### LINKED ISSUE #19: Testing: User Dashboard
Write tests to check all the ways that a user can interact with the dashboard.

### PR DESCRIPTION
**Pull request by rachellhill**
# Description

- Finishes cypress testing for user workflow 
- Adds error handling for bad network request 
- Cleans up code 
- Adjusts styling

## Issues

Closes #19 #34 #41 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor feature (non-breaking change that modifies existing work)
- [x] Cosmetic changes (changes to layout or visuals in CSS file)

## How Has This Been Tested?

- [ ] console.log()
- [x] dev tools
- [x] functionality check through web browser
- [ ] tested by viewing the markdown file preview
- [x] tested with testing framework

## Checklist:

- [x] My code follows the style guidelines of this project (semicolons, indentation, naming conventions, etc..)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have deleted all comments/console.log() after functionality and understanding is final
- [x] I have added any issues closed by this PR in the Issues section.
- [x] I have added others as reviewers

### CODE REVIEW

**Comment by whiteheadol** [comment]
Last official project PR of Turing!

File: src/components/App.js [hunk 1]
```diff
@@ -45,6 +49,8 @@ function App() {
         setThirtyDayAvgStraws={setThirtyDayAvgStraws}
         setThirtyDayAvgBags={setThirtyDayAvgBags}
         setThirtyDayAvgBottles={setThirtyDayAvgBottles}
+        isError={isError}
```
Comment (nbalmoori) [inline src/components/App.js:52]: good error handling here

File: cypress/e2e/userInterfaceTesting.cy.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    cy.get('.source-link').should('exist')
+  })
+  
+  it('Should display a 404 message if a user tries to visit a page that does not exist', () => {
+    cy.visit('http://localhost:3000/potato')
+    cy.get('.fof-box').contains(`Oh no - this page doesn't exist! Navigate to an existing page using the menu above.`)
+  })
+
+  it('Should be able to work the timer', () => {
+    cy.get('#start-btn').click().wait(5000)
```
Comment (nbalmoori) [inline cypress/e2e/userInterfaceTesting.cy.js:107]: glad we got this testing to work!

**Review by nbalmoori — APPROVED** [review]

File: cypress/e2e/userInterfaceTesting.cy.js (hunk 2 above)
Comment (whiteheadol) [inline cypress/e2e/userInterfaceTesting.cy.js:107]: I'm so glad we figured out timer testing!

**Review by whiteheadol — APPROVED** [review]

**Pull request merged**